src/core/server/saved_objects/migrationsv2/integration_tests/migration_7_13_0_unknown_types.test.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import Fs from 'fs';
+import Util from 'util';
+import { estypes } from '@elastic/elasticsearch';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import { Root } from '../../../root';
+
+const logFilePath = Path.join(__dirname, '7_13_unknown_types_test.log');
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+
+async function removeLogFile() {
+  // ignore errors if it doesn't exist
+  await asyncUnlink(logFilePath).catch(() => void 0);
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('migrates the documents to the highest version', async () => {
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+          // dataset contains 2 type of unknown docs
+          // `foo` documents
+          // `space` documents (to mimic a migration with disabled plugins)
+          dataArchive: Path.join(__dirname, 'archives', '7.13.0_with_unknown_so.zip'),
+        },
+      },
+    });
+
+    root = createRoot();
+
+    esServer = await startES();
+    await root.setup();
+
+    try {
+      await root.start();
+    } catch (err) {
+      const errorMessage = err.message;
+
+      expect(
+        errorMessage.startsWith(
+          'Unable to complete saved object migrations for the [.kibana] index: Migration failed because documents ' +
+            'from unknown types were found. To proceed with the migration, please delete these documents from the ' +
```

Comment:
Out of context of saved objects, if a user sees this message, they might not know what the 'type' is here. Maybe we can improve the message string to make it clear what the 'type' refers to?